hooks/useDetectOS.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+import { detectOS } from '../util/detectOS';
+import { getBitness } from '../util/getBitness';
+import type { UserOS } from '../types/userOS';
+
+export const useDetectOS = () => {
+  const [userOSData, setUserOSData] = useState<{
+    os: UserOS;
+    bitness?: number;
+  }>({ os: 'OTHER' });
+
+  useEffect(() => {
+    const os = detectOS();
+
+    if (os === 'WIN') {
```

Comment:
I find this logic really verbose. I rather not have special treatment for "WIN" os. As in theory the bitness API works on other OSs, even if we're not interested in.

Please if possible, remove this specific "WIN" logic (iof/else) statement, and `bitness` should always be there :)

Reply:
Actually I was conflicted about having the "special treatment" or not. :)
It is unnecessary for now (having or not having doesn't change anything in the codebase).
Agree, it is premature. For now, let me remove it.

Reply:
It's also about having a standardises API. macOS and Linux still have bitness because Linux can still be x86/x64 and macOS too (old versions I guess lol)

Reply:
I guess bitness doesn't matter outside Windows in general :)
Just googled about it more...